File: backend/EventosCba.API/Controllers/TicketsController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using EventosCba.API.Data;
using EventosCba.API.Models;
using Microsoft.EntityFrameworkCore;
using System.Security.Claims;

namespace EventosCba.API.Controllers
{
    [ApiController]
    [Route("api/[controller]")]
    [Authorize]
    public class TicketsController : ControllerBase
    {
        private readonly ApplicationDbContext _context;

        public TicketsController(ApplicationDbContext context)
        {
            _context = context;
        }

        [HttpPost("purchase")]
        public async Task<IActionResult> PurchaseTicket([FromBody] PurchaseRequest request)
        {
            var userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
            if (userId == null) return Unauthorized();

            var eventItem = await _context.Events.FindAsync(request.EventId);
            if (eventItem == null) return NotFound(new { message = "Event not found" });

            // Create ticket
            var ticket = new Ticket
            {
                EventId = request.EventId,
                UserId = userId,
                PricePaid = eventItem.Price,
                QRCode = Guid.NewGuid().ToString(), // Simple QR generation
                PurchasedAt = DateTime.UtcNow
            };

            _context.Tickets.Add(ticket);
            await _context.SaveChangesAsync();

            return Ok(new { message = "Ticket purchased successfully", ticketId = ticket.Id, qrCode = ticket.QRCode });
        }

        [HttpGet("my-tickets")]
        public async Task<IActionResult> GetMyTickets()
        {
            var userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
            if (userId == null) return Unauthorized();

            var tickets = await _context.Tickets
                .Where(t => t.UserId == userId)
                .Select(t => new
                {
                    t.Id,
                    t.EventId,
                    EventTitle = t.Event!.Title,
                    EventDate = t.Event.Date,
                    t.QRCode,
                    t.PricePaid,
                    t.IsUsed
                })
                .ToListAsync();

            return Ok(tickets);
        }
    }

    public class PurchaseRequest
    {
        public int EventId { get; set; }
    }
}

File: backend-node/src/controllers/ticketsController.js
const prisma = require('../config/database');
const crypto = require('crypto');

/**
 * @route   POST /api/tickets/purchase
 * @desc    Purchase a ticket for an event
 * @access  Private
 */
const purchaseTicket = async (req, res) => {
    try {
        const { eventId } = req.body;
        const userId = req.user.userId;

        // Verify event exists
        const event = await prisma.event.findUnique({
            where: { id: parseInt(eventId) }
        });

        if (!event) {
            return res.status(404).json({ message: 'Event not found' });
        }

        // Generate QR code (simple hash for now)
        const qrCode = crypto.randomBytes(16).toString('hex');

        // Create ticket
        const ticket = await prisma.ticket.create({
            data: {
                userId,
                eventId: parseInt(eventId),
                qrCode
            },
            include: {
                event: true
            }
        });

        res.status(201).json({
            message: 'Ticket purchased successfully!',
            ticket
        });
    } catch (error) {
        console.error('[Tickets] Purchase error:', error);
        res.status(500).json({ message: 'Error purchasing ticket' });
    }
};

/**
 * @route   GET /api/tickets/my-tickets
 * @desc    Get all tickets for authenticated user
 * @access  Private
 */
const getMyTickets = async (req, res) => {
    try {
        const userId = req.user.userId;

        const tickets = await prisma.ticket.findMany({
            where: { userId },
            include: {
                event: true
            },
            orderBy: { purchaseDate: 'desc' }
        });

        res.status(200).json(tickets);
    } catch (error) {
        console.error('[Tickets] Get my tickets error:', error);
        res.status(500).json({ message: 'Error fetching tickets' });
    }
};

module.exports = {
    purchaseTicket,
    getMyTickets
};
